src/object_store.cpp

```diff
@@ -517,6 +517,19 @@ void ObjectStore::verify_compatible_for_read_only(std::vector<SchemaChange> cons
     verify_no_errors<InvalidSchemaChangeException>(verifier, changes);
 }
 
+void ObjectStore::verify_compatible_for_read_only_alternative(std::vector<SchemaChange> const& changes)
+{
+    using namespace schema_change;
+    struct Verifier : SchemaDifferenceExplainer {
+        using SchemaDifferenceExplainer::operator();
+
+        void operator()(RemoveProperty) { }
```

Comment:
So what is the conclusion to solve this? Shall we have two modes `ReadonlyAlternative` and `ReadonlyStrict` where the `ReadonlyStrict` will throw if the schema doesn't match?